### PR TITLE
python3 migration: omicronxx driver

### PR DESCRIPTION
**Pull request by philipwinklerdelmic**

### CODE REVIEW

**Comment by philipwinklerdelmic** [comment]
done

File: src/odemis/driver/omicronxx.py [hunk 1]
```diff
@@ -143,43 +144,43 @@ def sendCommand(self, com):
         while True:
             line = self.readMessage()
             if line[0] == "$": # ad-hoc message => we don't care
-                logging.debug("Skipping ad-hoc message '%s'", line.encode('string_escape'))
+                logging.debug("Skipping ad-hoc message '%s'", line)
```
Comment (pieleric) [inline src/odemis/driver/omicronxx.py]: This one also needs to be escaped.

File: src/odemis/driver/omicronxx.py [hunk 2]
```diff
@@ -143,43 +144,43 @@ def sendCommand(self, com):
...
             char = self._serial.read()
-        logging.debug("Received: '%s'", line.encode('string_escape'))
+        logging.debug("Received: '%s'", line.decode('latin1', 'backslashreplace'))
 
         # Check it's a valid answer
         if not char: # should always finish by a "\r"
             raise IOError("Controller timeout.")
 
-        return line
+        return line.decode('latin1') if sys.version_info[0] >=3 else line
```
Comment (pieleric) [inline src/odemis/driver/omicronxx.py]: It should be fine to return unicode on Python 2. => remove the `if`.

File: src/odemis/driver/omicronxx.py [hunk 3]
```diff
@@ -385,18 +385,18 @@ def _setValue(self, com, val=None):
         ans = self.acc.sendCommand("%s%s%s" % (com, self._com_chan, val))
         if not ans.startswith(com):
             raise IOError("Expected answer to start with %s but got %s" %
-                          (com, ans.encode('string_escape')))
+                          (com, ans))
```
Comment (pieleric) [inline src/odemis/driver/omicronxx.py]: All the following lines should have "ans" escaped, as we don't know what it contains.

**Comment by philipwinklerdelmic** [comment]
done

File: src/odemis/driver/omicronxx.py (hunk 1 above)
Comment (pieleric) [reply src/odemis/driver/omicronxx.py]: Still needs `to_str_escape()` here.

**Comment by philipwinklerdelmic** [comment]
updated

File: src/odemis/driver/omicronxx.py [hunk 4]
```diff
@@ -367,8 +368,7 @@ def _getValue(self, com):
         fullcom = "%s%s" % (com, self._com_chan)
         ans = self.acc.sendCommand(fullcom)
         if not ans.startswith(fullcom):
-            raise IOError("Expected answer to start with %s but got %s" %
-                          (fullcom, ans.encode('string_escape')))
+            raise IOError("Expected answer to start with %s but got %s" % (fullcom, ans))
```
Comment (pieleric) [inline src/odemis/driver/omicronxx.py]: `to_str_escape(ans)` here.

File: src/odemis/driver/omicronxx.py [hunk 5]
```diff
@@ -435,7 +435,7 @@ def GetSpecInfo(self):
             wl = int(m.group("wl")) * 1e-9 # m
             power = int(m.group("power")) * 1e-3 # W
         except Exception:
-            raise ValueError("Failed to decode spec info answer '%s'" % ans.encode('string_escape'))
+            raise ValueError("Failed to decode spec info answer '%s'" % ans)
```
Comment (pieleric) [inline src/odemis/driver/omicronxx.py]: ` to_str_escape(ans)`

**Pull request merged**